src/main/resources/org/cbioportal/legacy/persistence/mybatis/MolecularDataMapper.xml

```diff
@@ -3,15 +3,40 @@
 
 <mapper namespace="org.cbioportal.legacy.persistence.mybatis.MolecularDataMapper">
 
+    <sql id="selectSingleMolecularProfile">
+        SELECT gp.genetic_profile_id
+        FROM genetic_profile gp
+        WHERE gp.stable_id = #{molecularProfileId}
+    </sql>
+    
+    <sql id="selectMultipleEntrezGeneIds">
+        SELECT g.genetic_entity_id
+        FROM gene g
+        WHERE g.entrez_gene_id IN
+            <foreach item="item" collection="entrezGeneIds" open="(" separator="," close=")">#{item}</foreach>
+    </sql>
+    
+    <sql id="whereInSingleMolecularProfile">
+        <where>
+            genetic_alteration.genetic_profile_id = (<include refid="selectSingleMolecularProfile"/>)
+            <if test="entrezGeneIds != null and !entrezGeneIds.isEmpty()">
+                AND genetic_alteration.genetic_entity_id IN (<include refid="selectMultipleEntrezGeneIds"/>)
+            </if>
+        </where>
+    </sql>
+    
     <sql id="whereInMultipleMolecularProfiles">
         <where>
             <if test="molecularProfileIds != null and !molecularProfileIds.isEmpty()">
-                genetic_profile.stable_id IN
-                    <foreach item="item" collection="molecularProfileIds" open="(" separator="," close=")">#{item}</foreach>
+                genetic_alteration.genetic_profile_id IN (
+                    SELECT gp.genetic_profile_id
+                    FROM genetic_profile gp
+                    WHERE gp.stable_id IN 
+                        <foreach item="item" collection="molecularProfileIds" open="(" separator="," close=")">#{item}</foreach>
+                )   
             </if>
             <if test="entrezGeneIds != null and !entrezGeneIds.isEmpty()">
-                AND gene.entrez_gene_id IN
-                    <foreach item="item" collection="entrezGeneIds" open="(" separator="," close=")">#{item}</foreach>
+                AND genetic_alteration.genetic_entity_id IN (<include refid="selectMultipleEntrezGeneIds"/>)
```

Comment:
It's confusing as hell that 'selectMultipleEntrezGeneIds' actually returns genetic_entity_ids.   What if we renamed this 'selectMultipleGeneticEntityIdsByEntrezGeneIds'.  I know that's a mouthful but otherwise, people will forever confused the entit ids with entrezids?

Reply:
I think it is okay to include `genetic_alteration.genetic_entity_id =` as well and rename to something more general purpose

Reply:
Included these in the SQL and renamed to be less confusing